File: test_fetch.swift
import Foundation
import EventKit

let store = EKEventStore()
let semaphore = DispatchSemaphore(value: 0)

if #available(macOS 14.0, *) {
    store.requestFullAccessToEvents { granted, error in
        semaphore.signal()
    }
} else {
    store.requestAccess(to: .event) { granted, error in
        semaphore.signal()
    }
}
_ = semaphore.wait(timeout: .now() + 2.0)

let now = Date()
let startDate = now.addingTimeInterval(-86400)
let endDate = now.addingTimeInterval(86400 * 2)

let predicate = store.predicateForEvents(withStart: startDate, end: endDate, calendars: nil)
let events = store.events(matching: predicate)

print("FOUND_EVENTS_COUNT:\(events.count)")
for ev in events {
    let title = ev.title ?? "No Title"
    let start = Int(ev.startDate.timeIntervalSince1970)
    let end = Int(ev.endDate.timeIntervalSince1970)
    let url = ev.url?.absoluteString ?? ""
    let location = ev.location ?? ""
    let notes = ev.notes ?? ""
    print("EV|\(title)|\(start)|\(end)|\(url)|\(location)|\(notes.replacingOccurrences(of: "\n", with: " "))")
}

File: test_fetch.py
import EventKit, Foundation
from datetime import datetime, timedelta

store = EventKit.EKEventStore.alloc().init()
now = datetime.now()
start = now - timedelta(days=30)
end = now + timedelta(days=30)

ns_start = Foundation.NSDate.dateWithTimeIntervalSince1970_(start.timestamp())
ns_end = Foundation.NSDate.dateWithTimeIntervalSince1970_(end.timestamp())

cals = store.calendarsForEntityType_(EventKit.EKEntityTypeEvent)
predicate = store.predicateForEventsWithStartDate_endDate_calendars_(ns_start, ns_end, cals)
events = store.eventsMatchingPredicate_(predicate)
print(f"Total events in last/next 30 days: {len(events) if events else 0}")
for e in (events or []):
    print(e.title(), e.startDate())
